src/reducers/starWarsReducer.js

```diff
@@ -1,13 +1,31 @@
-import /* we need our action types here*/ "../actions";
+import { FETCH_CHARACTER, FETCH_CHARACTER_SUCCESS, FETCH_CHARACTER_FAILURE } from '../actions';
+
 const initialState = {
-  characters: []
-  // Array characters, Boolean fetching, null error.
+  characters: [],
+  fetching: false,
+  error: null
 };
+
 export const charsReducer = (state = initialState, action) => {
+  console.log("reducer", action)
   switch (action.type) {
-    // Fill me in with the important reducers
-    // action types should be FETCHING, SUCCESS and FAILURE
-    // your switch statement should handle all of these cases.
+    case FETCH_CHARACTER:
+      return {
+        ...state,
+        fetching: true
+      }
+    case FETCH_CHARACTER_SUCCESS:
+      return {
+        ...state,
+        fetching: false,
+        characters: action.payload
+      }
+    case FETCH_CHARACTER_FAILURE:
+      return {
+        ...state,
+        fetching: false,
+        error: action.payload
```

Comment:
think about all of your cases here: you want to display the error if the fetch fails, but when you make another attempt, the error needs to be reset. in your `FETCH_CHARACTER` case, set error to null